searx/engines/livespace.py

```diff
@@ -0,0 +1,118 @@
+# SPDX-License-Identifier: AGPL-3.0-or-later
+"""
+ LiveSpace (Videos)
+"""
+
+from urllib.parse import quote_plus
+from json import loads
+from datetime import datetime
+
+# about
+about = {
+    "website": 'https://live.space',
+    "wikidata_id": None,
+    "official_api_documentation": None,
+    "use_official_api": True,
+    "require_api_key": False,
+    "results": 'JSON',
+}
+
+categories = ['videos']
+
+engine_url = 'https://backend.live.space/search/public/stream' + '?page={page}&searchKey={query}&size={size}'
+
+# engine dependent config
+paging = True
+number_of_results = 10
+
+
+def request(query, params):
+    # basic search
+    page = params['pageno'] - 1
+
+    string_args = dict(query=quote_plus(query), page=page, size=number_of_results)
+
+    params['url'] = engine_url.format(**string_args)
+
+    params['headers'] = {'Accept': 'application/json', 'Content-Type': 'application/json'}
+
+    return params
```

Comment:
```suggestion
base_url = 'https://backend.live.space'

# engine dependent config
paging = True
results_per_page = 10


def request(query, params):
    args = {
        'page': params['pageno'] - 1,
        'size': results_per_page,
        'query': query
    } 
    params['url'] = f"{base_url}/search/public/stream?{urlenode(args)}"
    params['headers'] = {'Accept': 'application/json', 'Content-Type': 'application/json'}

    return params
```

Explanation concerning renaming `number_of_results`: This is generally being used for the total found amount of results by a query, usually large numbers like 11_000_000.